llama_index/llms/openai_like.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Awaitable, Callable, Dict, Optional, Sequence
+
+from llama_index.bridge.pydantic import Field
+from llama_index.callbacks import CallbackManager
+from llama_index.llms import OpenAI
+from llama_index.llms.base import (
+    LLM,
+    ChatMessage,
+    ChatResponse,
+    ChatResponseAsyncGen,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseAsyncGen,
+    CompletionResponseGen,
+    LLMMetadata,
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.llms.generic_utils import (
+    achat_to_completion_decorator,
+    acompletion_to_chat_decorator,
+    astream_chat_to_completion_decorator,
+    astream_completion_to_chat_decorator,
+    chat_to_completion_decorator,
+    completion_to_chat_decorator,
+    stream_chat_to_completion_decorator,
+    stream_completion_to_chat_decorator,
+)
+from llama_index.llms.openai_utils import (
+    acompletion_with_retry,
+    completion_with_retry,
+    from_openai_message_dict,
+    to_openai_message_dicts,
+)
+
+
+class OpenAILike(OpenAI):
+    class_type = "openai_like"
+
+    context_window: int = Field(discription="Custom context_window for custom models.")
+    is_chat_model: bool = Field(
+        default=False, description="Indicats that the custom model is a chat_model."
+    )
+    is_function_calling_model: bool = Field(
+        default=False,
+        description="Indicats that the custom model is a function calling model.",
+    )
+
+    def __init__(self, **data: Any) -> None:
```

Comment:
Check out `LocalAI` one, basically without doing `self.context_window = context_window`, the pydantic serialization won't work

Reply:
Nevermind to this, this was incorrect.  If you're curious, check https://github.com/run-llama/llama_index/pull/7982, where I simplified `LocalAI.__init__` with this learning